internal/framework5provider/dynamic_edge_resource_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package framework
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+)
+
+// Ref: https://github.com/hashicorp/terraform-plugin-framework/issues/969
+// This test confirms that dynamic computed attributes are marked as unknown, both value AND type.
+// Dynamic computed attributes can change the type after plan in this scenario.
+func TestDynamicEdge_computed_type_changes(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		ProtoV5ProviderFactories: map[string]func() (tfprotov5.ProviderServer, error){
+			"framework": providerserver.NewProtocol5WithError(New()),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: `resource "framework_dynamic_edge" "test" {
+					required_dynamic = "value1"
+				}`,
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue("framework_dynamic_edge.test", tfjsonpath.New("required_dynamic"), knownvalue.StringExact("value1")),
+					// Created as a boolean
+					statecheck.ExpectKnownValue("framework_dynamic_edge.test", tfjsonpath.New("computed_dynamic_type_changes"), knownvalue.Bool(true)),
+				},
+			},
+			{
+				Config: `resource "framework_dynamic_edge" "test" {
+					required_dynamic = "new value"
+				}`,
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue("framework_dynamic_edge.test", tfjsonpath.New("required_dynamic"), knownvalue.StringExact("new value")),
+					// After update, it's a number!
+					statecheck.ExpectKnownValue("framework_dynamic_edge.test", tfjsonpath.New("computed_dynamic_type_changes"), knownvalue.Int64Exact(200)),
+				},
+			},
+		},
```

Comment:
From @bflad in https://github.com/hashicorp/terraform-plugin-framework/pull/970#discussion_r1539093182
> Do we want to add ConfigPlanChecks to verify its being marked as unknown in the plans?

Reply:
Apologies for the gnarly diff, but I added plan checks for unknowns while renaming the files cb83381 ⚡